test/suite/test_backup09.py

```diff
@@ -0,0 +1,151 @@
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+# test_backup09.py
+#   Verify opening a backup cursor forces a log file switch.
+#
+
+import os, shutil, stat
+import wiredtiger, wttest
+from wtscenario import make_scenarios
+
+class test_backup09(wttest.WiredTigerTestCase):
+    conn_config = 'config_base=false,create,log=(enabled)'
+    uri = 'table:coll1'
+    backup_dir = 'backup.dir'
+
+    types = [
+        # checkpoint: whether to explicitly checkpoint some data before opening
+        #   the backup cursor.
+        #
+        # all_log_files: whether to copy all files in the source directory, or
+        #   only the files returned from the backup cursor. Copying the
+        #   additional log files will result in more operations being recovered.
+        ('checkpoint', {'checkpoint': True, 'all_log_files': False}),
+        ('no_checkpoint', {'checkpoint': False, 'all_log_files': False}),
+        ('all_log_files', {'checkpoint': True, 'all_log_files': True}),
+    ]
+    scenarios = make_scenarios(types)
+
+    def data_and_start_backup(self):
+        self.session.create(self.uri, 'key_format=i,value_format=i')
+
+        cursor = self.session.open_cursor(self.uri)
+        doc_id = 0
+
+        for i in range(10):
+            doc_id += 1
+            cursor[doc_id] = doc_id
+
+        if self.checkpoint:
+            self.session.checkpoint()
+
+        for i in range(10):
+            doc_id += 1
+            cursor[doc_id] = doc_id
+
+        last_doc_in_backup = doc_id
+        self.assertEqual(1, len(filter(lambda x:
+            x.startswith('WiredTigerLog.'), os.listdir('.'))))
+        backup_cursor = self.session.open_cursor('backup:')
+        # This test asserts that writes after a checkpoint will not be part of
+        # the data/log files returned by a backup cursor that was opened prior
+        # to the checkpoint. Taking a checkpoint will also force a log write to
+        # a new file, due to the lazy-evaluated flag set by the backup cursor.
+        self.session.checkpoint()
```

Comment:
I think this checkpoint call may not be necessary. By setting the flag before `log_force_write` the logging subsystem will not lazy-evaluate it and will create the new file when it switches log slots.

Reply:
I had left the checkpoint in because the contract we need to provide is weaker (providing a relationship between backup cursors and writes only in the face of checkpoints). Though that contract is subtle and probably confusing to others compared to simply "backup cursors change the log file".

So I took this out, and one of the tests started failing. Back to debugging :)